module/vision/GoalDetector/src/GoalDetector.cpp

```diff
@@ -260,6 +262,100 @@ namespace module::vision {
                         }
                     }
 
+                    log<NUClear::DEBUG>("Culling colinear goals");
+
+                    for (auto it1 = goals->goals.begin(); it1 != goals->goals.end(); it1 = std::next(it1)) {
+                        log<NUClear::DEBUG>(fmt::format("Post Disances {}", it1->post.distance));
+                    }
+                    // If a sided goal is "behind" another sided goal it cannot be a goal
+                    // loop through all goals (sided), if any other goal is behind it (ie ~same phi theta && goal1_dist
+                    // < goal2_dist) throw it away! (or for now, make it an unknow goal or something)
+                    const Eigen::Affine3f Hcw(horizon.Hcw.cast<float>());
+                    for (auto goal_post_a = goals->goals.begin(); goal_post_a != goals->goals.end();) {
+                        bool increment_a = false;
+                        // Vector from camera to the bottom center of the post, projected to the observation
+                        Eigen::Matrix<float, 2, 1> rGOo_a =
+                            (goal_post_a->post.bottom.head<2>() * goal_post_a->post.distance);
+
+                        // Length of that vector
+                        const float distance_a = rGOo_a.norm();
+
+                        rGOo_a /= distance_a;
+
+                        for (auto goal_post_b = goals->goals.begin(); goal_post_b != goals->goals.end();) {
+
+                            // No point checking a post against itself
+                            if (goal_post_a != goal_post_b) {
+                                log<NUClear::DEBUG>(fmt::format("Goal post a at distance {}", distance_a));
+                                // Vector from camera to the bottom center of the post, projected to the observation
+                                Eigen::Matrix<float, 2, 1> rGOo_b =
+                                    (goal_post_b->post.bottom.head<2>() * goal_post_b->post.distance);
+
+                                // Length of that vector
+                                const float distance_b = rGOo_b.norm();
+
+                                rGOo_b /= distance_b;
+
+                                log<NUClear::DEBUG>(fmt::format("Goal post b at distance {}", distance_b));
+
+                                // Calculate the angular distance between the two posts
+                                const float post_to_post_theta = rGOo_a.dot(rGOo_b);
+
+                                const float d = distance_a < distance_b ? distance_a : distance_b;
+                                const float radius =
+                                    (2.0 * d)
```

Comment:
I have no idea how this calculation works lol

Reply:
This is the inverse of a calculation above